File: lesson17_Authentication/FabricMarket_BLL/Contracts/Identity/UserBriefModel.cs
using lesson11_FabricMarket_DomainModel.Models.Identity;

namespace FabricMarket_BLL.Contracts.Identity
{
    public class UserBriefModel
    {
        public required string FirstName { get; set; }
        public string? LastName { get; set; }
        public required string Email { get; set; }
        public required UserRoleEnum Role { get; set; }
    }
}

File: lesson22_Angular/hello/src/models/identity/UserBriefModel.ts
import { UserRoleEnum } from "./RoleEnum";

export class UserBriefModel{
    public static FullName(user: UserBriefModel): string {
        return  user.FirstName + ' ' + user.LastName;
    }

    constructor(
        public Email: string,
        public FirstName: string,
        public LastName: string,
        public Role: UserRoleEnum,
        public FullName: string,
    )
    {
    }
}